src/lib.rs

```diff
@@ -28,18 +28,22 @@
 //!     Ok(mfd)
 //! }
 //! ```
-
-#![deny(missing_docs)]
-
-extern crate either;
-extern crate libc;
-extern crate thiserror;
+#![deny(
+    missing_docs,
+    intra_doc_link_resolution_failure,
+    clippy::all,
+    unreachable_pub,
+    unused,
+)]
+#![cfg_attr(docsrs, feature(doc_cfg))]
```

Comment:
Still need to finish integrating this...

Reply:
`docs.rs` runs a nightly rustdoc by default, so any configuration that occurs here is in order to keep this library compiling on stable. In particular, all that’s really necessary for annotations to appear on documentation is this kind of attribute, then

```
#[cfg_attr(docsrs, doc(cfg(target_os="linux")))] 
```

on items we want to mark as linux-specific (pretty much all of this library) and finally a

```
[package.metadata.docs.rs]
rustdoc-args = ["--cfg", "docsrs"] 
```

in `Cargo.toml`.

Reply:
> docs.rs runs a nightly rustdoc by default

Ah, thanks, this was the bit I was missing.